src/java/org/apache/cassandra/service/accord/async/AsyncLoader.java

```diff
@@ -136,7 +137,7 @@ public boolean load(AsyncContext context, BiConsumer<Object, Throwable> callback
                 readFuture = referenceAndDispatchReads(context);
                 state = State.LOADING;
             case LOADING:
-                if (readFuture != null && !readFuture.isDone())
+                if (readFuture != null && (!readFuture.isDone() || !readFuture.isSuccess()))
```

Comment:
when the future failed we treat this as success without checking for success; this can cause the tx to hang

Reply:
this could just be `if (readFuture != null && !readFuture.isSuccess())`, since isSuccess returning true implies it is also done